OpenSim/Examples/Plugins/CoupledBushingForceExample/CoupledBushingForce.cpp

```diff
@@ -136,192 +96,62 @@ void CoupledBushingForce::copyData(const CoupledBushingForce &aForce)
-        errorMessage = "Invalid bushing body1 (" + _body1Name + ") specified in Force " + getName();
-        throw (Exception(errorMessage.c_str()));
-    }
-    if (!aModel.updBodySet().contains(_body2Name)) {
-        errorMessage = "Invalid bushing body2 (" + _body2Name + ") specified in Force " + getName();
-        throw (Exception(errorMessage.c_str()));
-    }
+    //Default frames
+    PhysicalOffsetFrame frame1;
+    PhysicalOffsetFrame frame2;
+    frame1.setName("offset_frame1");
+    frame2.setName("offset_frame2");
+
+    constructProperty_offset_frame1(frame1);
+    constructProperty_offset_frame2(frame2);
+    // default bushing material properties
+    // 6x6 stiffness matrix
+    constructProperty_stiffness_row1(Vec6(0));
+    constructProperty_stiffness_row2(Vec6(0));
+    constructProperty_stiffness_row3(Vec6(0));
+    constructProperty_stiffness_row4(Vec6(0));
+    constructProperty_stiffness_row5(Vec6(0));
+    constructProperty_stiffness_row6(Vec6(0));
+    // 6x6 damping matrix
+    constructProperty_damping_row1(Vec6(0));
+    constructProperty_damping_row2(Vec6(0));
+    constructProperty_damping_row3(Vec6(0));
+    constructProperty_damping_row4(Vec6(0));
+    constructProperty_damping_row5(Vec6(0));
+    constructProperty_damping_row6(Vec6(0));
 }
-
-void CoupledBushingForce::extendAddToSystem(SimTK::MultibodySystem& system) const
+ 
+void CoupledBushingForce::extendFinalizeFromProperties()
 {
-    Super::extendAddToSystem(system);
-
-    Body &body1 = _model->updBodySet().get(_body1Name);
-    Body &body2 = _model->updBodySet().get(_body2Name);
-
-    // Beyond the const Component get access to underlying SimTK elements
-    CoupledBushingForce* mutableThis = const_cast<CoupledBushingForce *>(this);
-
-    // Get underlying mobilized bodies
-    mutableThis->_b1 = &body1.getMobilizedBody();
-    mutableThis->_b2 = &body2.getMobilizedBody();
-    // Define the transforms for the bushing frames affixed to the specified bodies
-    SimTK::Rotation r1; r1.setRotationToBodyFixedXYZ(_orientationInBody1);
-    SimTK::Rotation r2; r2.setRotationToBodyFixedXYZ(_orientationInBody2);
-    // Hang on to the transforms for the bushing frames
-    mutableThis->_inb1 = SimTK::Transform(r1, _locationInBody1);
-    mutableThis->_inb2 = SimTK::Transform(r2, _locationInBody2);
-}
+    Super::extendFinalizeFromProperties();
 
-//=============================================================================
-// OPERATORS
-//=============================================================================
-//_____________________________________________________________________________
-/**
- * Assignment operator.
- *
- * @return Reference to this object.
- */
-CoupledBushingForce& CoupledBushingForce::operator=(const CoupledBushingForce &aForce)
-{
-    Force::operator=(aForce);
-    copyData(aForce);
-    return(*this);
+    //mark the two PhysicalOffsetFrames as subcomponents 
+    addComponent(&upd_offset_frame1());
```

Comment:
I'm confused about why these `addComponent()` calls are in finalizeFromProperties. Are these what we could eliminate if object properties were automatically made into subcomponents?